### PR TITLE
strictNullChecks: locator, renderers, more

### PR DESCRIPTION
**Pull request by fregante**
## Checklist

- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

File: src/hooks/useMemoCompare.ts [hunk 1]
```diff
@@ -41,7 +41,8 @@ function useMemoCompare<T>(
   });
 
   // Finally, if equal then return the previous value
-  return isEqual ? previous : next;
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion, @typescript-eslint/no-unnecessary-type-assertion -- Can't be undefined if it's equal to T
```
Comment (grahamlangford) [inline src/hooks/useMemoCompare.ts:44]: Technically if `useMemoCompare` is used in a file that hasn't been strict null checked yet, next could still be null | undefined and `undefined === undefined`.

Comment (fregante) [reply src/hooks/useMemoCompare.ts:44]: Technically every strict null checked file can still pass around undefined due to the non-strict part.

In your scenario, if the input `T` is silently `string | undefined`, then `previous` still matches `T`, so no need for this type to be `T | undefined` (i.e. unnecessarily widened)

**Review by grahamlangford — APPROVED** [review]
LGTM. See failing test and merge conflict.

File: src/background/proxyUtils.ts [hunk 2]
```diff
@@ -57,7 +57,8 @@ export function proxyResponseToAxiosResponse(
     return {
       data: data.json,
       status: data.status_code,
-      statusText: data.reason ?? data.message,
+      statusText:
+        data.reason ?? data.message ?? safeGuessStatusText(data.status_code),
```
Comment (twschiller) [inline src/background/proxyUtils.ts:61]: 👍

**Comment by twschiller** [comment]
I fixed the test - there's some inconsistent nullish-ness of the `root` property floating around the runtime

File: src/data/service/requestErrorUtils.test.ts [hunk 3]
```diff
@@ -24,7 +24,7 @@ describe("safeGuessStatusText", () => {
   });
 
   it("doesn't throw error on invalid code", () => {
-    expect(safeGuessStatusText(-200)).toBeNull();
+    expect(safeGuessStatusText(-200)).toBe("Unknown HTTP status");
```
[the remark this answers is not in the document]
Comment (twschiller) [reply src/data/service/requestErrorUtils.test.ts]: Updated to unknown to be less ambiguous. The status code is know (we know the number), we just don't know the status text that corresponds to it

**Pull request merged**